api4/websocket.go

```diff
@@ -16,23 +20,47 @@ func (api *API) InitWebSocket() {
 }
 
 func connectWebSocket(c *Context, w http.ResponseWriter, r *http.Request) {
-	upgrader := websocket.Upgrader{
-		ReadBufferSize:  model.SOCKET_MAX_MESSAGE_SIZE_KB,
-		WriteBufferSize: model.SOCKET_MAX_MESSAGE_SIZE_KB,
-		CheckOrigin:     c.App.OriginChecker(),
+	fn := c.App.OriginChecker()
+	if fn != nil && !fn(r) {
+		c.Err = model.NewAppError("origin_check", "api.web_socket.connect.check_origin.app_error", nil, "", http.StatusBadRequest)
+		return
+	}
+
+	upgrader := ws.HTTPUpgrader{
+		Timeout: 5 * time.Second,
```

Comment:
This looks like an addition. Just wondering if we actually need to set this.

Reply:
Yeah, the gorilla implementation doesn't actually implement any timeout while writing to a network connection. So it's good to have this.

Reply:
Interesting. I thought this was only about the handshake process and that it essentially matched [`HandshakeTimeout`](https://pkg.go.dev/github.com/gorilla/websocket#Upgrader) which we never used, hence my question.

Reply:
Oh no you are right. That timeout does exactly that. I must have looked at somewhere else. It's just that we weren't using it.